generator/grammar.py

```diff
@@ -19,13 +19,14 @@
 
 dec_literal = Word(nums)
 hex_literal = Combine("0x" + Word(hexnums))
+signed_literal = Combine(Optional('-') + (dec_literal ^ hex_literal))
```

Comment:
Why not add a parse action here?

Reply:
Good point. While we do not use parse actions much in the current grammar we could. Might be worth considering as a generic grammar cleanup